File: 60_Permutation_Sequence.py
class Solution(object):
    def getPermutation(self, n, k):
        """
        :type n: int
        :type k: int
        :rtype: str
        """
        # O(n) : 80%
        num = "123456789"
        res = ""
        fn = [1]*n
        for i in range(1, n):
            fn[i] = fn[i - 1] * i
        k = k - 1

        for i in range(1, n + 1):
            j = k // fn[n - i]
            k = k % fn[n - i]
            res = res + num[j]
            num = num[:j] + num[j + 1:]
        return res

solution = Solution()
print solution.getPermutation(3, 3)

File: 60_Permutation_Sequence.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution{
public:
    string getPermutation(int n, int k) {
        /*O(n) : 80% */
        string num = "123456789";
        string res;
        vector<int> fn(n, 1);
        for (int i = 1; i <= n; ++i) fn[i] = fn[i - 1] * i;
        k--;
        //for (int s : fn) cout << s << " ";
        for (int i = n; i > 0; --i){
            int j = k / fn[i - 1];
            k = k % fn[i - 1];
            res.push_back(num[j]);
            num.erase(j, 1);
        }
        return res;
    }
};

int main(){
    Solution* solution = new Solution();
    cout << solution->getPermutation(3, 3) << "\n";
    return 0;
}